java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/JavaCheckVerifier.java

```diff
@@ -206,23 +213,54 @@ public static void verifyIssueOnFile(String filename, String message, JavaFileSc
 
   private static void scanFile(String filename, JavaFileScanner check, JavaCheckVerifier javaCheckVerifier) {
     Collection<File> classpath = Lists.newLinkedList();
-    File testJars = new File(javaCheckVerifier.testJarsDirectory);
-    if (testJars.exists()) {
-      classpath = FileUtils.listFiles(testJars, new String[]{"jar", "zip"}, true);
+    Path testJars = Paths.get(javaCheckVerifier.testJarsDirectory);
+    if (Files.exists(testJars)) {
+      classpath = getFilesRecursively(testJars, new String[] {"jar", "zip"});
     } else if (!DEFAULT_TEST_JARS_DIRECTORY.equals(javaCheckVerifier.testJarsDirectory)) {
-      Fail.fail("The directory to be used to extend class path does not exists (" + testJars.getAbsolutePath() + ").");
+      Fail.fail("The directory to be used to extend class path does not exists (" + testJars.toAbsolutePath() + ").");
     }
     classpath.add(new File("target/test-classes"));
     scanFile(filename, check, javaCheckVerifier, classpath);
   }
 
+  private static List<File> getFilesRecursively(Path root, final String[] extensions) {
+    final List<File> files = new ArrayList<>();
+
+    FileVisitor<Path> visitor = new SimpleFileVisitor<Path>() {
+      @Override
+      public FileVisitResult visitFile(Path filePath, BasicFileAttributes attrs) throws IOException {
+        for (String extension : extensions) {
+          if (filePath.toString().endsWith("." + extension)) {
+            files.add(filePath.toFile());
+            break;
+          }
+        }
+        return FileVisitResult.CONTINUE;
+      }
+
+      @Override
+      public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+        return FileVisitResult.CONTINUE;
+      }
+    };
+
+    try {
+      Files.walkFileTree(root, visitor);
+    } catch (IOException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)